### PR TITLE
Fix #74246 redirect-ads.com

### LINKED ISSUE #74246: www3.f2movies.to
### Issue URL (Incorrect Blocking)
 [https://www3.f2movies.to/watch-movie/forgotten-8720.2324398](https://adguardteam.github.io/AnonymousRedirect/redirect.html?url=https%3A%2F%2Fwww3.f2movies.to%2Fwatch-movie%2Fforgotten-8720.2324398)

### Screenshots
<details>
  <summary>Screenshot 1</summary>

  Screenshot 1
</details>
<details>
  <summary>Screenshot 2</summary>

  Screenshot 2
</details>

### System configuration

Information | value
--- | ---
AdGuard product: | AdGuard Browser extension v3.5.31
Browser: | Chrome
Filters: | <b>Ad Blocking:</b><br/> AdGuard Base

### PR DESCRIPTION
**Pull request by AdamWr**
Issue - https://github.com/AdguardTeam/AdguardFilters/issues/74246

Fix popups from `hydrax/redirect-ads.com` video player.

Code - https://gist.github.com/AdamWr/4b51c6b7c2d7bd22ba2e14f5d3ffe0b4

In this case, `redirect-ads.com#%#//scriptlet("prevent-window-open")` can't be used, because it breaks website.
It's due to `win.focus();` part from their script:
```js
var win = window.open("test");
win.focus();
setTimeout(() => {
  if (win && !win.closed) {
    // [...]
  }
}, 50);

### CODE REVIEW

**Comment by slavaleleka** [comment]
`redirect-ads.com#%#//scriptlet('prevent-window-open', '1', '', '{focus=noopFunc}')` helps me with it

**Comment by AdamWr** [comment]
`redirect-ads.com#%#//scriptlet('prevent-window-open', '1', '', '{focus=noopFunc}')`
It breaks video player on my end.

**Comment by AdamWr** [comment]
<details><summary>Screenshot</summary>

image

</details>

The same happens on http://example.org/ with this rule:
`example.org,redirect-ads.com#%#//scriptlet('prevent-window-open', '1', '', '{focus=noopFunc}')`
if I run this code:
```js
var win = window.open("test");
win.focus();
setTimeout(() => {
  if (win && !win.closed) {
    console.log("Test");
  }
}, 50);
```

<details><summary>Screenshot</summary>

image

</details>

**Comment by ameshkov** [comment]
@slavaleleka interesting, shouldn't we implement some changes to `prevent-window-open`?

File: EnglishFilter/sections/general_extensions.txt [hunk 1]
```diff
@@ -247,7 +247,9 @@ videowood.tv#%#//scriptlet("abort-on-property-read", "_wm")
 videowood.tv#%#//scriptlet("abort-on-property-read", "PopURL")
 videowood.tv#%#//scriptlet("abort-on-property-read", "LieDetector")
 ! https://github.com/AdguardTeam/AdguardFilters/issues/74246
+! https://gist.github.com/AdamWr/4b51c6b7c2d7bd22ba2e14f5d3ffe0b4
 redirect-ads.com#%#//scriptlet("json-prune", "*", "banner_id")
+redirect-ads.com#%#(function(){if("function"===typeof Proxy){var a=/^./,c=function(){var b=document.createElement("iframe");b.src=a;b.style.setProperty("height","1px","important");b.style.setProperty("width","1px","important");b.style.setProperty("position","absolute","important");b.style.setProperty("top","-99999px","important");document.body.appendChild(b);setTimeout(function(){return b.remove()},1E3);return b};window.open=new Proxy(window.open,{apply:function(b,g,e){if(!0!==a.test(e[0]))return b.apply(g,e);b=c.contentWindow;return b=new Proxy(self,{get:function(f,d,h){return"closed"===d?!1:"opener"===d?window:"frameElement"===d?null:"function"===typeof Reflect.get(f,d)?function(){}:f[d]},set:function(){return Reflect.set.apply(Reflect,arguments)}})}})}})();
```
Comment (ameshkov) [inline EnglishFilter/sections/general_extensions.txt:252]: Please add a comment here:
`! TODO: replace with "prevent-window-open" scriptlet after https://github.com/AdguardTeam/Scriptlets/issues/117`

**Review by ameshkov — CHANGES_REQUESTED** [review]

File: EnglishFilter/sections/general_extensions.txt (hunk 1 above)
Comment (AdamWr) [reply EnglishFilter/sections/general_extensions.txt:253]: Done.

**Review by ameshkov — APPROVED** [review]

**Review by Alex-302 — APPROVED** [review]

**Pull request merged**